programs/system/src/invoke/sol_compression.rs

```diff
@@ -31,10 +31,8 @@ pub fn compress_or_decompress_lamports<
 ) -> Result<()> {
     if inputs.is_compress {
         compress_lamports(inputs, ctx)
-    } else if inputs.compress_or_decompress_lamports.is_some() {
-        decompress_lamports(inputs, ctx)
     } else {
-        Ok(())
+        decompress_lamports(inputs, ctx)
     }
 }
```

Comment:
this is now by default always going to execute `decompress_lamports()`,no? 
`is_compress` is not optional, so the default usage is is_compress = false, compress_or_decompress_lamports = None.

Reply:
The function yes but we only execute the function when `compress_or_decompress_lamports` is some.
Hence, this function will never be executed if we don't want to compress or decompress.